### DISCUSSION TITLE
Add function

### DISCUSSION THREAD

**Discussion by DuanhaoranCC** [pyg-team/pytorch_geometric#6679] [General]
Hi, I suggest adding remove_edges() and remove_nodes() to PyG's toolkit. I think the user is free to specify which nodes to delete. This is a commonly used function.

I am facing a problem now to delete a K-hop subgraph, but I have a little trouble and need your help.

```python
from torch_geometric.utils import k_hop_subgraph, subgraph

edge_index = torch.tensor([[0, 1, 2, 3, 4, 5], [2, 2, 4, 4, 6, 6]])
edge_attr = torch.arange(7)
subset, edge_index1, mapping, edge_mask = k_hop_subgraph(6, 2, edge_index, relabel_nodes=True)
mask = torch.ones(7)
mask[subset] = 0
edge_index, edge_attr = subgraph(mask.bool(), edge_index, edge_attr, relabel_nodes=True, num_nodes=7)
```

Error: IndexError: The shape of the mask [6] at index 0 does not match the shape of the indexed tensor [7] at index 0.

This seems intuitive, but the program crashes, do you have any suggestions?

**Comment by rusty1s** [comment]
Note that `edge_index.size(1) == edge_attr.size(0)`, so it needs to be:
```python
edge_attr = torch.arange(6)
```
